### PR TITLE
Cdn on edge tutorial

### PR DESCRIPTION
**Pull request by dynamite-bud**
Added a tutorial on **How to deploy a CDN for static assets** on wasmer edge

### CODE REVIEW

**Comment by dynamite-bud** [comment]
Add a note in the tutorial about how big the files (the upper limits) can be or if any files are not allowed ?

File: pages/edge/tutorials/_meta.json [hunk 1]
```diff
@@ -1,4 +1,5 @@
 {
   "react-static-site": "React Static Site",
-  "python-flask-server": "Python Flask Server"
+  "python-flask-server": "Python Flask Server",
+  "cdn-on-wasmer-edge": "CDN on Wasmer Edge"
```
Comment (syrusakbary) [inline pages/edge/tutorials/_meta.json:4]: This should be `Wasmer Edge as CDN` (and the url should be just `cdn`)

File: pages/edge/tutorials/cdn-on-wasmer-edge.mdx [hunk 2]
```diff
@@ -0,0 +1,357 @@
+import { Steps, Callout, FileTree } from "nextra-theme-docs";
+import StaticSite from "@components/deploy/quickstart/StaticSite";
+import edgeReactImg from "@assets/deploy/tutorials/edge-react.png";
+import Install from "@components/install.mdx";
+import Login from "@components/login.mdx";
+import CliVersionCallout from "@components/deploy/CliVersionCallout.mdx";
+
+# Creating a CDN on Wasmer Edge
```
Comment (syrusakbary) [inline pages/edge/tutorials/cdn-on-wasmer-edge.mdx:8]:
```suggestion
# Using Wasmer Edge as a CDN
```

File: pages/edge/tutorials/cdn-on-wasmer-edge.mdx [hunk 3]
```diff
@@ -0,0 +1,357 @@
...
+
+In this tutorial, we will create a CDN on Wasmer Edge. We'll be using the [`static-web-server`]()
```
Comment (syrusakbary) [inline pages/edge/tutorials/cdn-on-wasmer-edge.mdx:10]:
```suggestion
In this tutorial, we be demonstrate the CDN capabilities Wasmer Edge. For that, we'll be using the [`static-web-server`](WHERE THIS LINKS??)
```

File: pages/edge/tutorials/cdn-on-wasmer-edge.mdx [hunk 4]
```diff
@@ -0,0 +1,357 @@
...
+```bash
+$ mkdir config
+$ touch config/config.toml
+```
+
+You can copy the following `config.toml`.
+
+```toml
+[general]
+
```
Comment (syrusakbary) [inline pages/edge/tutorials/cdn-on-wasmer-edge.mdx:94]: We don't need to showcase every detail here, nor that big of a config. Please summarize it to the essential things only

File: pages/edge/tutorials/cdn-on-wasmer-edge.mdx [hunk 5]
```diff
@@ -0,0 +1,357 @@
...
+name = "<your-namespace>/<your-cdn-name>"
+version = "0.1.0"
+description = "<your-namespace>/<your-cdn-name> CDN on Wasmer Edge"
+
+[dependencies]
+"wasmer/static-web-server" = "^1"
+
+[fs]
+public = "public"
+config = "config" # 👈🏼 Add the config directory
```
Comment (syrusakbary) [inline pages/edge/tutorials/cdn-on-wasmer-edge.mdx:256]: We are using non absolute paths here. We should be using absolute paths

File: pages/edge/tutorials/cdn-on-wasmer-edge.mdx [hunk 6]
```diff
@@ -0,0 +1,357 @@
...
+
+[[command]]
+name = "script"
+module = "wasmer/static-web-server-async:webserver"
+runner = "https://webc.org/runner/wasi"
+
+[command.annotations.wasi] # 👈🏼 Add config.toml as an argument for the webserver
+main-args = ["-w", "./config/config.toml"]
```
Comment (syrusakbary) [inline pages/edge/tutorials/cdn-on-wasmer-edge.mdx:264]: Absolute paths

File: pages/edge/tutorials/cdn-on-wasmer-edge.mdx [hunk 7]
```diff
@@ -0,0 +1,357 @@
...
+</Callout>
+
+<Callout type="info" emoji="🌐">
+  You'll need to do `wasmer deploy` again if you add any new files to the
+  `public` directory.
+</Callout>
+
+<Callout type="warning" emoji="🚧">
+  The url of your CDN will be different from the one mentioned above.
+</Callout>
```
Comment (syrusakbary) [inline pages/edge/tutorials/cdn-on-wasmer-edge.mdx:345]: There's not a "Your CDN". You don't build a CDN, you publish INTO a CDN. This needs rephrasing

File: pages/edge/tutorials/cdn-on-wasmer-edge.mdx [hunk 8]
```diff
@@ -0,0 +1,357 @@
...
+
+</Steps>
+
+## Conclusion
+
+In this tutorial, we learned how to create a CDN on Wasmer Edge using the `static-web-server` package.
```
Comment (syrusakbary) [inline pages/edge/tutorials/cdn-on-wasmer-edge.mdx:351]: No, you can't create a CDN, you use Wasmer Edge as a CDN

File: pages/edge/tutorials/cdn-on-wasmer-edge.mdx [hunk 9]
```diff
@@ -0,0 +1,357 @@
...
+Waiting for package to become available.....
+Package '<your-namespace>/<your-cdn-name>@0.1.0' published successfully!
+
+Deploying app <your-namespace>-<your-cdn-name>...
+
+ ✅ App <your-namespace>/<your-namespace>-<your-cdn-name> was successfully deployed!
+
+> App URL: https://dynamite-bud-sws-cdn.wasmer.app
+> Versioned URL: https://2qjh9iec55z4.id.wasmer.app
+> Admin dashboard: https://wasmer.io/apps/dynamite-bud-sws-cdn
```
Comment (syrusakbary) [inline pages/edge/tutorials/cdn-on-wasmer-edge.mdx:324]: This url should not have anything about your user here. It should also be visitable by people

File: pages/edge/tutorials/_meta.json (hunk 1 above)
Comment (dynamite-bud) [reply pages/edge/tutorials/_meta.json]: changed that

File: pages/edge/tutorials/cdn-on-wasmer-edge.mdx (hunk 2 above)
Comment (dynamite-bud) [reply pages/edge/tutorials/cdn-on-wasmer-edge.mdx]: changed that

File: pages/edge/tutorials/cdn-on-wasmer-edge.mdx (hunk 3 above)
Comment (dynamite-bud) [reply pages/edge/tutorials/cdn-on-wasmer-edge.mdx]: Updated that

File: pages/edge/tutorials/cdn-on-wasmer-edge.mdx (hunk 5 above)
Comment (dynamite-bud) [reply pages/edge/tutorials/cdn-on-wasmer-edge.mdx]: I changed the paths to:
```toml
public = "/public"
config = "/config"
```

I think public is a good name for static assets ? As there's no application per-se but rather just hosting.

File: pages/edge/tutorials/cdn-on-wasmer-edge.mdx (hunk 8 above)
Comment (dynamite-bud) [reply pages/edge/tutorials/cdn-on-wasmer-edge.mdx]: Changed that

File: pages/edge/tutorials/cdn-on-wasmer-edge.mdx (hunk 7 above)
Comment (dynamite-bud) [reply pages/edge/tutorials/cdn-on-wasmer-edge.mdx]: Rephrased to: `The url of your app will be different from the one mentioned above.`

File: pages/edge/tutorials/cdn-on-wasmer-edge.mdx (hunk 6 above)
Comment (dynamite-bud) [reply pages/edge/tutorials/cdn-on-wasmer-edge.mdx]: changed that to use the path `/config/config.toml`

Comment (syrusakbary) [reply pages/edge/tutorials/cdn-on-wasmer-edge.mdx]: Nope, it should be `/app/config.toml`

File: pages/edge/tutorials/cdn-on-wasmer-edge.mdx (hunk 5 above)
Comment (syrusakbary) [reply pages/edge/tutorials/cdn-on-wasmer-edge.mdx]: Nope, Is better to do `"/app" = .`. That way you have both `/app/public` and `/app/config.toml` automatically on it

File: pages/edge/tutorials/wasmer-edge-as-cdn.mdx [hunk 10]
```diff
@@ -0,0 +1,357 @@
...
+
+The above command will give you the following prompt:
+
+```bash
+App type:
+> Static website # Select Static website
+  HTTP server
+  Browser shell
+  JS Worker (experimental)
+  Python Worker
```
Comment (syrusakbary) [inline pages/edge/tutorials/wasmer-edge-as-cdn.mdx:46]: Why are we still saying Python Worker? We should use the latest nomenclature

File: pages/edge/tutorials/wasmer-edge-as-cdn.mdx [hunk 11]
```diff
@@ -0,0 +1,357 @@
...
+Who should own this package?: wasmer
```
Comment (syrusakbary) [inline pages/edge/tutorials/wasmer-edge-as-cdn.mdx:47]: <THE OWNER>

File: pages/edge/tutorials/wasmer-edge-as-cdn.mdx [hunk 12]
```diff
@@ -0,0 +1,357 @@
...
+
+<Callout type="warning" emoji="🚧">
+  The url of your app will be different from the one mentioned above.
+</Callout>
+
+</Steps>
+
+## Conclusion
+
+In this tutorial, we learnt how to use Wasmer Edge as a CDN using the `static-web-server` package for hosting your static assets.
```
Comment (syrusakbary) [inline pages/edge/tutorials/wasmer-edge-as-cdn.mdx:351]: *learned (learnt is only more common in UK, not US)

**Pull request merged**